core/src/main/java/org/elasticsearch/common/util/concurrent/SizeBlockingQueue.java

```diff
@@ -35,7 +35,7 @@
 public class SizeBlockingQueue<E> extends AbstractQueue<E> implements BlockingQueue<E> {
 
     private final BlockingQueue<E> queue;
-    private final int capacity;
+    protected volatile int capacity;
```

Comment:
Maybe we could make `capacity` private again and have size-blocking queue rely on `#capacity()` which can be protected and overridable in resizable-blocking queue where the backing `capacity` field can be volatile. You might think it's an odd change, removing a volatile read for a method invocation, but this will be a biomorphic call site so the JIT can inline it.